resolver/addressable_resolver.go

```diff
@@ -56,6 +58,28 @@ func NewURIResolver(ctx context.Context, callback func(types.NamespacedName)) *U
 		},
 	}
 
+	ret.listerFactory = func(gvr schema.GroupVersionResource) (cache.GenericLister, error) {
+		_, l, err := ret.informerFactory.Get(ctx, gvr)
+		return l, err
+	}
+
+	return ret
+}
+
+// NewURIResolver constructs a new URIResolver with context and a callback
+// for a given listableType (Listable) passed to the URIResolver's tracker.
+// extra when controller context is used
+func NewURIResolverWithControllerContext(ctx context.Context, dmInformerFactory pkgapisduck.InformerFactory,
```

Comment:
Hmm, the name of this functions seems kinda misleading. Why `WithControllerContext`? `NewURIResolver` seems to have changed its meaning of the context as well so maybe this is more like `NewURIResolverWithFactory` or something like that?

Reply:
Oh! Yes I think you're right @markusthoemmes, which answers my question in https://github.com/knative/pkg/pull/2095#discussion_r613030091 -- we already did fix the existing constructor. But now I'm not sure if/why we need the second ctor at all? 😊